core/src/main/java/org/infinispan/cache/impl/EncoderCache.java

```diff
@@ -1018,6 +1032,237 @@ public B setValue(B value) {
       }
    }
 
+   private static class KeyFunctionEncoder<I, O> implements Function<Publisher<I>, O>, InjectableComponent {
```

Comment:
The `EncoderCache` is already a large file. What do you think of moving these classes outside?

Reply:
Sure